### ISSUE TITLE
ComplexMethod false positive for returning anonymous inner class

### ISSUE THREAD

**Issue by TWiStErRob** [arturbosch/detekt#1037]
## Expected Behavior
No detection.

## Current Behavior
```
ComplexMethod - 11/10 - [f] at  ...
```

## Steps to Reproduce (for bugs)
Run `ComplexMethod` on this:
```kotlin
fun f(): List<Any> {
    return object : List<Any> {
        override val size: Int get() = TODO("not implemented")

        override fun contains(element: Any): Boolean {
            TODO("not implemented")
        }

        override fun containsAll(elements: Collection<Any>): Boolean {
            TODO("not implemented")
        }

        override fun get(index: Int): Any {
            TODO("not implemented")
        }

        override fun indexOf(element: Any): Int {
            TODO("not implemented")
        }

        override fun isEmpty(): Boolean {
            TODO("not implemented")
        }

        override fun iterator(): Iterator<Any> {
            TODO("not implemented")
        }

        override fun lastIndexOf(element: Any): Int {
            TODO("not implemented")
        }

        override fun listIterator(): ListIterator<Any> {
            TODO("not implemented")
        }

        override fun listIterator(index: Int): ListIterator<Any> {
            TODO("not implemented")
        }

        override fun subList(fromIndex: Int, toIndex: Int): List<Any> {
            TODO("not implemented")
        }
    }
}
```

## Context
The method is quite simple:
```kotlin
fun f(): List<Any> {
    return object : List<Any> { ... }
}
```

## Your Environment

* Version used: 1.0.0.RC7-3
* Operating System and version: Windows 10
* Link to your project: private but full repro provided above

**Comment by arturbosch** [comment]
Seeing this again, I'm not sure if we just skip nested function as they are visited by themselfs and reports on demand?
